src/templates/static/special/rebootLocalHost.yml

```diff
@@ -0,0 +1,21 @@
+# Reboots a system if any of the preceeding plays sets the 'insights_needs_reboot' variable to true.
+# The variable can be overridden to suppress this behavior.
+- name: Reboot localhost
+  hosts: "@@HOSTS@@"
+  vars:
+    insights_signature_exclude: "/hosts"
+  become: True
+  gather_facts: False
+  tasks:
+    - when:
+        - insights_needs_reboot is defined
+        - insights_needs_reboot
+      block:
+        - name: Reboot system
+          shell: sleep 2 && shutdown -r +1 now "Ansible triggered reboot"
+          async: 1
+          poll: 0
+          ignore_errors: true
+        - name: Schedule insights-client after boot
```

Comment:
Doing things in this order creates a potential for race condition (host gets rebooted before the file gets touched). It is very unlikely but it may still be better to trigger reboot after everything has been set up.